apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php

```diff
@@ -31,6 +35,42 @@ function should_show_coming_soon_page() {
 	return apply_filters( 'a8c_show_coming_soon_page', $should_show );
 }
 
+/**
+ * Determines whether the coming soon page should be bypassed.
+ *
+ * It checks if share code is provided as GET parameter, or as a cookie.
+ * Then it validates the code against blog option, and if sharing code is valid,
+ * it allows bypassing the Coming Soon page.
+ *
+ * Finally, it sets a code in cookie and sets header that prevents robots from indexing.
+ *
+ * @return bool
+ */
+function is_accessed_by_valid_share_link() {
+	if ( isset( $_GET['share'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$share_code = $_GET['share']; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	} elseif ( isset( $_COOKIE['share_code'] ) ) {
+		$share_code = $_COOKIE['share_code'];
+	} else {
+		$share_code = '';
+	}
+
+	$preview_links_option = get_option( 'wpcom_public_preview_links' );
+
+	if ( ! is_array( $preview_links_option ) || ! count( $preview_links_option ) || ! $share_code ) {
+		return false;
+	}
+
+	if ( ! in_array( $share_code, array_column( $preview_links_option, 'code' ), true ) ) {
+		return false;
+	}
+
+	setcookie( 'share_code', $share_code, time() + 3600, '/', false, is_ssl() );
```

Comment:
It seems a little bit odd to have a `is_*` function _also_ set a cookie. Should this be set higher up the stack?

Also, as it relates to the cookie... Batcache (full page cache) ignores all cookies by default (p1666205398542069-slack-C7YPW6K40). Only very special cookies are bypassed. We'll want to make sure our cookie is named in such a manner that it bypasses cache (such that the PHP code can check to see what should be displayed).

Reply:
@danielbachhuber, just for my information, I think that batcache allows cookies starting with `wp` or `wordpress`, correct?
So naming the cookie `wp_share_code` will do the trick right?

Reply:
@kozer I believe so, yes. It would benefit from testing in any case 😁

Reply:
@danielbachhuber I extracted that logic out of `is_*` function.

Thanks for the notes about Batcache, I will review this part more.

Reply:
I've renamed the cookie as recommended and tested it with the Atomic dev site - Batcache doesn't save those requests.